modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/services/AlarmReceiver.java

```diff
@@ -17,29 +17,76 @@
  */
 package org.wso2.emm.agent.services;
 
+import android.net.Uri;
 import android.os.AsyncTask;
 import org.wso2.emm.agent.AndroidAgentException;
+import org.wso2.emm.agent.R;
 import org.wso2.emm.agent.utils.Constants;
+import org.wso2.emm.agent.utils.Preference;
 
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.util.Log;
+import android.widget.Toast;
 
 /**
  * This class is a broadcast receiver which triggers on local notification timeouts.
  */
 public class AlarmReceiver extends BroadcastReceiver {
 
 	private static final String TAG = AlarmReceiver.class.getName();
+	private String operation = null;
 
 	@Override
 	public void onReceive(Context context, Intent intent) {
 		if (Constants.DEBUG_MODE_ENABLED) {
 			Log.d(TAG, "Recurring alarm; requesting alarm service.");
 		}
-		OperationTask operationTask = new OperationTask();
-		operationTask.execute(context);
+
+		if (intent.hasExtra(context.getResources().getString(R.string.alarm_scheduled_operation))) {
+			operation = intent.getStringExtra(context.getResources().getString(R.string.alarm_scheduled_operation));
+
+			if(operation != null && operation.trim().equals(Constants.Operation.INSTALL_APPLICATION)) {
+				Preference.putString(context, context.getResources().getString(R.string.alarm_schedule), null);
+				Preference.putString(context, context.getResources().getString(R.string.app_uri), null);
+				Toast.makeText(context, "App install request initiated by admin.",
+						Toast.LENGTH_SHORT).show();
+				//Prepare for install
+				String packageUri;
+				if (intent.hasExtra(context.getResources().getString(R.string.app_uri))) {
+					packageUri = intent.getStringExtra(context.getResources().getString(R.string.app_uri));
```

Comment:
EMM agent's AlarmReceiver is independent from that of System service. First the ApplicationManager APIs get called and there happens the system app availability check and then the operation is pushed to either Alarms.